lib/src/license_detection/token_matcher.dart

```diff
@@ -0,0 +1,353 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:pana/src/license_detection/license.dart';
+
+/// Range of tokens in input text that matched to a range of tokens in known license.
+@sealed
+class MatchRange {
+  /// Range of tokens that were found to be a match in input text.
+  Range input;
+
+  /// Range of tokens that were found to be a match in known license.
+  Range source;
+
+  /// Number of tokens that were matched in this range.
+  int tokensClaimed;
+
+  @visibleForTesting
+  MatchRange(
+    this.input,
+    this.source,
+    this.tokensClaimed,
+  );
+}
+
+/// Indicates the start and end index for a range of tokens or diffs.
+@sealed
+class Range {
+  /// Start index of the token in this range.
+  int start;
+
+  /// End index(exclusive) of the token in this range.
+  int end;
+
```

Comment:
Can range have a helper method like:
```dart
int get length => end - start;
```

I imagine that sometimes it could be useful that have...